src/Billing/AutoBilling.php

```diff
@@ -0,0 +1,320 @@
+        if (empty($hasBilling['encounter'])) {
+            if (!empty($code)) {
+                    $codetype = "CPT4";
+                    $this->insertCPTBilling($event_date, $code, $this->pid, $provider, $this->userid, $enc, $text, $fees, $icd10, $codetype, $modifier);
+            }
+
+            //Enter ICD10 if it exist
+            if (!empty($desc) && empty($res)) {
+                    $codetype = "ICD10";
+                    $this->insertICDBilling($event_date, $codetype, $icd10, $this->pid, $provider, $this->userid, $enc, $desc);
+            }
+        }
+    }// End of billing entries
+
+    /**
+     * @param $event_date
+     * @param $code
+     * @param $pid
+     * @param $provider
+     * @param $userid
+     * @param $enc
+     * @param $text
+     * @param $fees
+     * @param $icd
+     * @param $codetype
+     * @param $modifier
+     */
+    private function insertCPTBilling($event_date, $code, $pid, $provider, $userid, $enc, $text, $fees, $icd10, $codetype, $modifier)
+    {
+        $sql = "REPLACE INTO billing SET " .
+            "date = ? , " .
+            "code_type = ?, " .
+            "code = ?, " .
+            "pid = ?, " .
+            "provider_id = ?, " .
+            "user = ?, " .
+            "groupname = 'default', " .
+            "authorized = '1', " .
+            "encounter = ?, " .
+            "code_text = ?, " .
+            "billed = '0', " .
+            "activity = '1', " .
+            "modifier = ?, " .
+            "units = '1', " .
+            "fee = ?, " .
+            "justify = ?, " .
+            "pricelevel = 'standard'";
+
+             $justify = "ICD10|$icd10:";
+              sqlStatementThrowException($sql, [$event_date, $codetype, $code, $pid, $provider, $userid, $enc, $text, $modifier, $fees, $justify]);
+
+        return 1;
+    }
+
+    /**
+     * @param $enDate
+     * @param $icd10
+     * @param $pid
+     * @param $provider
+     * @param $userid
+     * @param $enc
+     * @param $desc
+     */
+    private function insertICDBilling($event_date, $codetype, $icd10, $pid, $provider, $userid, $enc, $desc)
+    {
+        $sql = "REPLACE INTO billing SET "
+            . "date = ?,"
+            . "code_type = ?,"
+            . "code = ?,"
+            . "pid = ?,"
+            . "provider_id = ?,"
+            . "user = ?,"
+            . "groupname = 'default',"
+            . "authorized = '1',"
+            . "encounter = ?,"
+            . "code_text = ?,"
+            . "activity = '1',"
+            . "units = '1', "
+            . "fee = '0.00'";
+
+            sqlStatementThrowException($sql, [$event_date, $codetype, $icd10, $pid, $provider, $userid, $enc, $desc]);
+    }
+
+    /**
+     * @param $pid
+     * @return mixed
+     */
+    public function getDiagnosis()
+    {
+        /**
+         * get the most recent diagnosis
+         */
+        $sql = "select title, diagnosis, type from lists where pid = ? ORDER BY id DESC LIMIT 1";
```

Comment:
We store medications in lists.  Will this create problems for you here that you aren't excluding types such as medication or immunization?

Reply:
I have used this code in other projects and that particular function is not used by anything at the moment. It is a dormant code. It can be enhanced to bring back specific types of list information. However, right now it does nothing.

Reply:
Looking this over, It seems that the hasBilling() function can be moved to the AutoBilling class. What are you thoughts on that?
Thanks for the compliment. Trying to work my way onto your team.

Reply:
You're talking about the hasBilling in C_FormSOAP? Yea it makes more sense to put it in the AutoBilling class.

Reply:
It produces this when executed. This will work well for providers that don't use a lot of codes and use the SOAP note. 
![image](https://user-images.githubusercontent.com/834958/134755378-384fef81-404d-4513-8b8c-bf12f5801654.png)
This way they can do the note and billing on one page!